x/emissions/keeper/msgserver/msg_server_requests.go

```diff
@@ -123,5 +149,10 @@ func (ms msgServer) RequestInference(ctx context.Context, msg *types.MsgRequestI
 	if err != nil {
 		return nil, err
 	}
+	// 13. Activate topic
+	err = ms.ActivateTopic(ctx, request.TopicId)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
If you want to activate the topic after the `RequestInference` is processed, you can do so by directly calling the keeper method. If `ActivateTopic` is not a method that is externally accessible, it doesn't make sense to keep it as a message in this specific case. Also, you will save an unnecessary query `GetTopicUnmetDemand` since you already have the BidAmount in the request struct.

Reply:
The reason I used `GetParamsMinTopicUnmetDemand` is that in case of BidAmount can be not enough for meeting demand at first. So current request's BidAmount is less than `MinTopicUnmetDemand` but `topicUnmetDemand` meet for demand. In this case we can active topic?

Reply:
yeah a bid could have less than `MinTopicUnmetDemand`. That should not block the request from being submitted